internal/operators/ocs/validations.go

```diff
@@ -152,23 +164,30 @@ func (o *ocsValidator) ValidateOCSRequirements(cluster *models.Cluster) bool {
 	err = setOperatorStatus(cluster, status)
 	if err != nil {
 		o.log.Error("Failed to set Operator status ", err)
-		return false
+		return validationFailure
 	}
 
-	return canDeployOCS
+	if canDeployOCS {
+		return validationSuccess
+	}
+	return validationFailure
 }
 
-func (o *ocsValidator) nodeResources(host *models.Host, cpuCount *int64, totalRAM *int64, diskCount *int64, hostsWithDisks *int64, insufficientHosts *[]string) error {
+func (o *ocsValidator) nodeResources(host *models.Host, cpuCount *int64, totalRAM *int64, diskCount *int64, hostsWithDisks *int64, insufficientHosts *[]string) (bool, error) {
 	var inventory models.Inventory
-	if err := json.Unmarshal([]byte(host.Inventory), &inventory); err != nil {
+	// if inventory is empty
+	if host.Inventory == "" {
+		o.log.Info("Empty Inventory of host with hostID ", *host.ID)
+		return true, nil // to indicate that inventory is empty and the ValidationStatus must be Pending
+	} else if err := json.Unmarshal([]byte(host.Inventory), &inventory); err != nil {
```

Comment:
What could cause this to fail? Invalid inventory?

Reply:
yes, when inventory is not populated correctly.

Reply:
Should that be treated as equivalent to not having the inventory? Can the cluster recover from this state?

Reply:
the cluster can recover from that when the host is in discovering/ insufficient status the host will run the inventory command again.

Reply:
In that case, should both cases - inventory exists and inventory is incomplete/incorrect - be treated the same way?

Reply:
@gamli75 , do we need to return error, if the inventory is not populated correctly? Or we need to set status as `Missing inventory` ?

Reply:
I think we should keep it as you handled it - it's a different case than an empty string